src/frontend/src/utils/with_options.rs

```diff
@@ -60,33 +61,42 @@ impl WithOptions {
     pub fn new(inner: HashMap<String, String>) -> Self {
         Self {
             inner: inner.into_iter().collect(),
+            ref_secret: Default::default(),
         }
     }
 
-    pub fn from_inner(inner: BTreeMap<String, String>) -> Self {
-        Self { inner }
+    pub fn from_inner(inner: HashMap<String, String>) -> Self {
+        Self {
+            inner,
+            ref_secret: Default::default(),
+        }
     }
 
     /// Get the reference of the inner map.
-    pub fn inner(&self) -> &BTreeMap<String, String> {
+    pub fn inner(&self) -> &HashMap<String, String> {
         &self.inner
     }
 
-    pub fn inner_mut(&mut self) -> &mut BTreeMap<String, String> {
+    pub fn inner_mut(&mut self) -> &mut HashMap<String, String> {
         &mut self.inner
     }
 
     /// Take the value of the inner map.
-    pub fn into_inner(self) -> BTreeMap<String, String> {
+    pub fn into_inner(self) -> HashMap<String, String> {
         self.inner
     }
```

Comment:
such a definition seem a bit dangerious because the `ref_secret` could may be forgotten

Reply:
+1, please consider we may add other catalogs to refer to

Reply:
What do you mean could may be forgotten? 

secret should not be part of inner. It's in `ref_secret: HashMap<String, ObjectName>,`

Reply:
1. This function consumes self but only returns a part of it.
2. I think the common meaning of into_inner in Rust is not to because the internal variable called "inner"... https://doc.rust-lang.org/std/?search=into_inner

Reply:
I see. This will need further refactoring.